certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/UpdateCertificateService.java

```diff
@@ -0,0 +1,52 @@
+package se.inera.intyg.certificateservice.application.certificate.service;
+
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.certificateservice.application.certificate.dto.UpdateCertificateRequest;
+import se.inera.intyg.certificateservice.application.certificate.dto.UpdateCertificateResponse;
+import se.inera.intyg.certificateservice.application.certificate.service.validation.UpdateCertificateRequestValidator;
+import se.inera.intyg.certificateservice.application.common.ActionEvaluationFactory;
+import se.inera.intyg.certificateservice.domain.certificate.model.CertificateId;
+import se.inera.intyg.certificateservice.domain.certificate.service.UpdateCertificateDomainService;
+
+@Service
+@RequiredArgsConstructor
+public class UpdateCertificateService {
+
+  private final UpdateCertificateRequestValidator updateCertificateRequestValidator;
+  private final UpdateCertificateDomainService updateCertificateDomainService;
+  private final ElementDataConverter elementDataConverter;
+  private final ActionEvaluationFactory actionEvaluationFactory;
+  private final CertificateConverter certificateConverter;
+
+  public UpdateCertificateResponse update(UpdateCertificateRequest updateCertificateRequest,
+      String certificateId) {
+    updateCertificateRequestValidator.validate(updateCertificateRequest, certificateId);
+
+    final var actionEvaluation = actionEvaluationFactory.create(
+        updateCertificateRequest.getPatient(),
+        updateCertificateRequest.getUser(),
+        updateCertificateRequest.getUnit(),
+        updateCertificateRequest.getCareUnit(),
+        updateCertificateRequest.getCareProvider()
+    );
+
+    final var elementDataList = updateCertificateRequest.getCertificate().getData()
+        .entrySet()
+        .stream()
```

Comment:
A thought is to filter any data that don't need to be converted.. for example filter away any data that is of category. Instead of filter nonNull afterwards.

Reply:
I like it, done!